inference_client/client.py

```diff
@@ -22,24 +22,33 @@ def __init__(
         """
 
         try:
-            self._auth_token = login(token)
+            self._auth_token = login(token) if token else None
```

Comment:
Why `if token else None`?

Reply:
OK. But for now, if the user does not provide a token, we rely on hubble sdk to authenticate by popping out a browser window.

Reply:
Yes, the token can be get from browser. But why the user should do it if the token is never used in self-hosting.

Reply:
Actually, I'm not fan of the browser behavior. It does not work when I want to use the client in other system, e.g., FastAPI.

Reply:
The inference-client is not a cli, it is a package to be used in other systems.

Reply:
I would suggest abandoning the login behavior.